### PR TITLE
Get LeaderBoard error

### PR DESCRIPTION
**Pull request by AMihneaa**

### CODE REVIEW

File: server/Game.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        }
+    }
+
+    async getLeaderboard (token: string) : Promise<typeLeaderBoardResponse> {
+        try {
+            
+            const isValid = await this.checkSession(token);
+
+            if (isValid.status == "error"){
+                throw new Error("Invalid Session!");
```
Comment (cristim67) [inline server/Game.ts:26]: you need to return an object with status error and msgerror :)

File: server/Game.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            }
+
+            const table = await this.prisma.leaderboard.findMany();
+
+            if (!table){
+                throw new Error ("Leaderboard is invalid!");
```
Comment (cristim67) [inline server/Game.ts:32]: same here

File: server/Game.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            }
+
+            const rows = table.map( (event) => {
```
Comment (cristim67) [inline server/Game.ts:35]: here you need to declare a type

File: server/Game.ts [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        if (this.prisma){
+            console.log("Connected");
+        }
+        else {
+            console.log("Database error");
+        }
+    }
+
+    async getLeaderboard (token: string) : Promise<typeLeaderBoardResponse> {
+        try {
```
Comment (cristim67) [inline server/Game.ts:21]: remove this block. when you call the db function you need to put a catch function and return null if you have an error, after check the variable if null.

File: server/Game.ts [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+                status: "ok",
+                
+            }
+        }catch(error){
+            return {
+                status: "error",
+                errorMessage: `${error}`
+            }
+        }
+    }
```
Comment (cristim67) [inline server/Game.ts:67]: same here.

File: server/Game.ts [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+    async checkSession (token: string) : Promise<SessionResponse>{
+        try {
+            if (!token){
+                throw new Error("Token is invalid");
```
Comment (cristim67) [inline server/Game.ts:72]: same here

File: server/Game.ts [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+            }
+
+            const activeSession = await this.prisma.session.findFirst({
+                where: { token: token.slice(1, token.length - 1) },
+              });
+
+            if (!activeSession){
+                throw new Error("Session expired!");
```
Comment (cristim67) [inline server/Game.ts:80]: same here

File: server/Game.ts [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+            if (isValid.status == "error"){
+                throw new Error(isValid.errorMessage);
+            }
+
+            const question = await this.prisma.questions.findFirst({where: {id: questionId}}).catch( (error) => {
+                console.log(error);
+                throw new Error ("Invalid Question!");
+            })
+
+            question?.answer?.map ( (event) => {
```
Comment (cristim67) [inline server/Game.ts:73]: should have type

File: server/Game.ts [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+    }
+
+    async checkSession (token: string) : Promise<SessionResponse>{
+        try {
+            if (!token){
+                throw new Error("Token is invalid");
+            }
+
+            const activeSession = await this.prisma.session.findFirst({
+                where: { token: token.slice(1, token.length - 1) },
```
Comment (cristim67) [inline server/Game.ts:100]: i think you can pass just token.

File: server/Game.ts [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+        try {
+            const isValid = await this.checkSession(token);
+
+            if (isValid.status == "error"){
+                throw new Error (isValid.errorMessage);
+            }
+
+            const table = await this.prisma.quickRound.findMany();
+
+            table.map( (event) => {
```
Comment (cristim67) [inline server/Game.ts:157]: event should have type

File: server/models/typeGame.ts [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+
+export type ResponseGetNumberQuestions = {
+    status: string,
+    numberQuestions?: number,
+    errorMessage?: string,
+}
+
+export type GetAnswersQuickRound = {
+    status: string, 
+    answers?: [],
```
Comment (cristim67) [inline server/models/typeGame.ts:23]: this should look like this Answers[] where Answers is:

```
export type Answers = {
   answer1: string,
   score1: number,
   answer2: string,
   score2: number,
   answer3: string,
   score3: number,
   answer4: string,
   score4: number,
   answer5: string,
   score5: number,
   answer6: string,
   score6: number,
}
```

File: server/Game.ts [hunk 12]
```diff
@@ -0,0 +1,197 @@
...
+            }
+        } catch (error) {
+            return {
+                status: "error",
+                errorMessage: `${error}`
+            }
+        }
+    }
+    
+    async updateQuickRound (ceva: string[] ,token: string) : Promise<UpdateQuickRoundResponse> {
```
Comment (cristim67) [inline server/Game.ts:175]: the method should be like this:

`async updateQuickRound (input: Answers[],token: string) : Promise<UpdateQuickRoundResponse>`

where type Answers is:

`export type Answers = { 
answer1: string, 
score1: number, 
answer2: string, 
score2: number,
answer3: string, 
score3: number, 
answer4: string, 
score4: number, 
answer5: string, 
score5: number, 
answer6: string, 
score6: number, 
}`

Comment (AMihneaa) [reply server/Game.ts:175]: Wouldn't it be better to do the following? 
```
export type Answers = {
  answers: { answer: string, score: number }[];
};
```

File: server/models/typeGame.ts (hunk 11 above)
Comment (AMihneaa) [reply server/models/typeGame.ts:23]: Wouldn't it be better to do the following?

```
export type Answers = {
  answers: { answer: string, score: number }[];
};
```

**Review by MDGcode — APPROVED** [review]

**Review by lseIT — APPROVED** [review]

**Pull request merged**